src/app/modules/gb-analysis-module/fractalis-components/gb-fractalis-chart/gb-fractalis-chart.component.ts

```diff
@@ -1,37 +1,128 @@
-import {Component, Input, OnInit} from '@angular/core';
+/**
+ * Copyright 2017 - 2018  The Hyve B.V.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {AfterViewInit, Component, Input} from '@angular/core';
 import {Chart} from '../../../../models/chart-models/chart';
 import {ChartType} from '../../../../models/chart-models/chart-type';
 import {FractalisService} from '../../../../services/fractalis.service';
+import {FractalisChartDescription} from '../../../../models/fractalis-models/fractalis-chart-description';
+import {FractalisData} from '../../../../models/fractalis-models/fractalis-data';
 import {Concept} from '../../../../models/constraint-models/concept';
+import {ChartValidator} from '../../../../utilities/chart-validator';
 
 @Component({
   selector: 'gb-fractalis-chart',
   templateUrl: './gb-fractalis-chart.component.html',
   styleUrls: ['./gb-fractalis-chart.component.css']
 })
-export class GbFractalisChartComponent implements OnInit {
+export class GbFractalisChartComponent implements AfterViewInit {
 
   @Input() chart: Chart;
+  private fractalisChartDescription: FractalisChartDescription;
+  private fractalisChart: object;
+  private chartValidator: ChartValidator;
```

Comment:
looks like ChartValidator is a utility, doesn't have to live inside a chart component

Reply:
hmm, but validation is done per each chart separately. ChartValidator is not only a helper class with static methods, but also has a fields specific to each chart - `minCatVarLength`, `maxNumVarLength` etc (based on what is returned fom Fractalis), that is why each chart has an own ChartValidator instance. Maybe it could be named better, do you have some ideas?

Reply:
see my previous comment in chart.ts